### PR TITLE
Add Foundation Firefox Colors to photon_colors.xml

### PR DESCRIPTION
**Pull request by brampitoyo**
Color reference: https://www.figma.com/file/DyIIHvRgqt2EXfAVn1gV9c/Fx-Colors

Tagging @mcarare, as it may fix a few UX issues on Firefox Focus. Feel free to modify this PR as necessary.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by mcarare** [comment]
@brampitoyo Thanks for the updates. I have fixed the color naming for `photonInk05`, it was named `photonInk10`.

**Review by pocmo — COMMENTED** [review]
We somehow need to get those into `PhotonColors` for Jetpack Compose too:
https://github.com/mozilla-mobile/android-components/blob/main/components/ui/colors/src/main/java/mozilla/components/ui/colors/PhotonColors.kt

@Mugurell Do I remember correctly you wanted to update PhotonColors anyways? Was this for the same colors?

**Comment by Mugurell** [comment]
> We somehow need to get those into `PhotonColors` for Jetpack Compose too:
> https://github.com/mozilla-mobile/android-components/blob/main/components/ui/colors/src/main/java/mozilla/components/ui/colors/PhotonColors.kt
> 
> @Mugurell Do I remember correctly you wanted to update PhotonColors anyways? Was this for the same colors?

Yes. This are pretty much the same colors I was looking to add in https://github.com/mozilla-mobile/fenix/pull/21193. 
Now all the things converge to update the PhotonColors with these values. I can add a separate commit for this.

File: components/ui/colors/src/main/res/values/photon_colors.xml [hunk 1]
```diff
@@ -12,119 +12,184 @@
         https://design.firefox.com/photon/visuals/color.html
```
Comment (Mugurell) [inline components/ui/colors/src/main/res/values/photon_colors.xml:12]: @brampitoyo The new color values are different from the ones in https://design.firefox.com/photon/visuals/color.html
Will the website be updated soon or should we change point future readers to another link?

**Comment by Mugurell** [comment]
I've added all the new xml colors but `Magenta`, `Teal` and `Grey` to `PhotonColors`.
Leaving for @brampitoyo to confirm Mihai's changes in commit 2 and 3. (updated values also used in `PhotonColors`).

File: components/ui/colors/src/main/java/mozilla/components/ui/colors/PhotonColors.kt [hunk 2]
```diff
@@ -14,98 +14,143 @@ import androidx.compose.ui.graphics.Color
  */
 @Suppress("MagicNumber") // It's a kind of magic ✨
```
Comment (MozillaNoah) [inline components/ui/colors/src/main/java/mozilla/components/ui/colors/PhotonColors.kt:15]: 😂

**Review by MozillaNoah — APPROVED** [review]

**Comment by Mugurell** [comment]
Seeing the confirmation from Bram on slack that these are the right colors - from Figma I'm going to merge this.

**Comment by Archaeopteryx** [comment]
@Mugurell This causes a [test failure](https://treeherder.mozilla.org/logviewer?job_id=351636792&repo=android-components&lineNumber=646) in `browser-menu`:
```
  TEST: classic highlight effect converts background tint
    [Robolectric] mozilla.components.browser.menu.BrowserMenuHighlightTest.classic highlight effect converts background tint: sdk=28; resources=binary
    I/MonitoringInstr: Instrumentation started!
  FAILURE

java.lang.AssertionError: expected:<HighPriorityHighlightEffect(backgroundTint=-65479)> but was:<HighPriorityHighlightEffect(backgroundTint=-38283)>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at mozilla.components.browser.menu.BrowserMenuHighlightTest.classic highlight effect converts background tint(BrowserMenuHighlightTest.kt:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:601)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:260)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:130)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:84)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:154)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy5.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:121)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by Mugurell** [comment]
> @Mugurell This causes a [test failure](https://treeherder.mozilla.org/logviewer?job_id=351636792&repo=android-components&lineNumber=646) in `browser-menu`:

Created https://github.com/mozilla-mobile/android-components/issues/10990 to address this.

**Comment by jonalmeida** [comment]
@mergifyio backport releases/93.0

Needed in Focus 93, so uplifting.

**Pull request merged**